### PR TITLE
eval,parser: implement binary-| eval

### PR DESCRIPTION
**Pull request by sbinet**
Fixes neugram/ng#196.

### CODE REVIEW

File: typecheck/typecheck_test.go [hunk 1]
```diff
@@ -94,12 +94,13 @@ var typeTests = []typeTest{
 			{"b", tipe.Float64},
 		},
 	},
-	{
-		[]string{
-			`a := [|]int64{{|"Col1","Col2"|}, {1, 2}, {3, 4}}`,
-		},
-		[]identType{{"a", &tipe.Table{tipe.Int64}}},
-	},
+	//	TODO:
```
Comment (crawshaw) [inline typecheck/typecheck_test.go]: What happened here? Was the test never working? I don't see how this change could have broken it.

Comment (sbinet) [reply typecheck/typecheck_test.go]: by correctly handling the binary `|` operator, it did break parsing `{|` and `|}`.
I do have in [sbinet/tableliteral](https://github.com/neugram/ng/commit/7d795a6c6efc763b8f51f85c275c3992d547d457) a change that reenables this test.

I introduced 2 new tokens: `{|` and `|}`

**Comment by sbinet** [comment]
I've rebased this PR on top of #213 
PTAL.

**Comment by crawshaw** [comment]
LGTM

**Pull request merged**